opensearch-service-domain-cdk/lib/stack-composer.ts

```diff
@@ -0,0 +1,181 @@
+        const advancedOptions = getContextForType('advancedOptions', 'object')
+        const useUnsignedBasicAuth = getContextForType('useUnsignedBasicAuth', 'boolean')
+        const fineGrainedManagerUserARN = getContextForType('fineGrainedManagerUserARN', 'string')
+        const fineGrainedManagerUserName = getContextForType('fineGrainedManagerUserName', 'string')
+        const fineGrainedManagerUserSecretManagerKeyARN = getContextForType('fineGrainedManagerUserSecretManagerKeyARN', 'string')
+        const cognitoIdentityPoolId = getContextForType('cognitoIdentityPoolId', 'string')
+        const cognitoRoleARN = getContextForType('cognitoRoleARN', 'string')
+        const cognitoUserPoolId = getContextForType('cognitoUserPoolId', 'string')
+        const customEndpointDomainName = getContextForType('customEndpointDomainName', 'string')
+        const customEndpointCertificateARN = getContextForType('customEndpointCertificateARN', 'string')
+        const customEndpointHostedZoneId = getContextForType('customEndpointHostedZoneId', 'string')
+        const enforceHTTPS = getContextForType('enforceHTTPS', 'boolean')
+        const noneToNodeEncryptionEnabled = getContextForType('nodeToNodeEncryptionEnabled', 'boolean')
+        const encryptionAtRestEnabled = getContextForType('encryptionAtRestEnabled', 'boolean')
+        const encryptionAtRestKmsKeyARN = getContextForType("encryptionAtRestKmsKeyARN", 'string')
+        const ebsEnabled = getContextForType('ebsEnabled', 'boolean')
+        const ebsIops = getContextForType('ebsIops', 'number')
+        const ebsVolumeSize = getContextForType('ebsVolumeSize', 'number')
+        const loggingAppLogEnabled = getContextForType('loggingAppLogEnabled', 'boolean')
+        const loggingAppLogGroupARN = getContextForType('loggingAppLogGroupARN', 'string')
+        const loggingAuditLogEnabled = getContextForType('loggingAuditLogEnabled', 'boolean')
+        const loggingAuditLogGroupARN = getContextForType('loggingAuditLogGroupARN', 'string')
+        const loggingSlowIndexLogEnabled = getContextForType('loggingSlowIndexLogEnabled', 'boolean')
+        const loggingSlowIndexLogGroupARN = getContextForType('loggingSlowIndexLogGroupARN', 'string')
+        const loggingSlowSearchLogEnabled = getContextForType('loggingSlowSearchLogEnabled', 'boolean')
+        const loggingSlowSearchLogGroupARN = getContextForType('loggingSlowSearchLogGroupARN', 'string')
+        const snapshotAutomatedStartHour = getContextForType('snapshotAutomatedStartHour', 'number')
+        const vpcId = getContextForType('vpcId', 'string')
+        const vpcSecurityGroupIds = getContextForType('vpcSecurityGroupIds', 'object')
+        const vpcSubnetIds = getContextForType('vpcSubnetIds', 'object')
+        const enableVersionUpgrade = getContextForType('enableVersionUpgrade', 'boolean')
+
+        // Check for partially configured features
+        if ((cognitoIdentityPoolId || cognitoRoleARN || cognitoUserPoolId) &&
```

Comment:
How do you know that those 3 values are required for Cognito auth?  How do you know that you've got all of the possible required groupings?  (maybe you don't, and it's worth calling that out)

Reply:
Yeah, I have concerns about this too.  I think the right approach is to provide a thin wrapper around the AWS APIs and let the CDK/CFN surface errors to the user if present.  Don't think we want to insert ourselves into that process as it won't be sustainable.

Reply:
This is based off requirements that the Domain CDK construct has on required fields for certain objects/features. I don't have much attachment here and am okay to remove these checks. From the construct side, some objects have required fields that should only be created if we have the required fields otherwise CDK will fail synthesis and produce some very ambiguous error messages such as `Cannot read property 'length' of undefined` for missing domain endpoint name field. To accommodate this, I feel we have to take on the required field assumption and check for this, but open to other work arounds